### PR TITLE
✨ Pic 1690 - match multiple defendants

### PR DESCRIPTION
**Pull request by jhevans**

### CODE REVIEW

**Comment by jhevans** [comment]
I managed to get the probation status updated working, at least according to the unit tests, but writing around the integration test is a little trickier and I'm not going to be able to get that in before CoP today. If it's ok I'll add a note to add it when we pick up [PIC-1713](https://dsdmoj.atlassian.net/browse/PIC-1713). We'd have to rewrite the test when we do this change anyway so will save a bit off effort overall.

File: src/main/java/uk/gov/justice/probation/courtcasematcher/restclient/model/offendersearch/MatchRequest.java [hunk 1]
```diff
@@ -70,8 +66,8 @@ public MatchRequest buildFrom(String pnc, Name fullName, LocalDate dateOfBirth)
             return builder.build();
         }
 
-        public MatchRequest buildFrom(CourtCase courtCase) throws IllegalArgumentException {
-            return buildFrom(courtCase.getFirstDefendant().getPnc(), courtCase.getFirstDefendant().getName(), courtCase.getFirstDefendant().getDateOfBirth());
+        public MatchRequest buildFrom(Defendant firstDefendant) throws IllegalArgumentException {
```
Comment (chris-faulkner-mojdt) [inline src/main/java/uk/gov/justice/probation/courtcasematcher/restclient/model/offendersearch/MatchRequest.java:69]: `firstDefendant` should be named `defendant` ?

File: src/main/java/uk/gov/justice/probation/courtcasematcher/restclient/CourtCaseRestClient.java [hunk 2]
```diff
@@ -73,14 +64,9 @@
...
     }
 
-    @Override
-    public Mono<Void> postMatches(String courtCode, String caseNo, GroupedOffenderMatches offenderMatches) {
-        return legacyCourtCaseRestClient.postMatches(courtCode, caseNo, offenderMatches);
-    }
-
-    @Override
-    public Mono<Void> postOffenderMatches(String caseId, String defendantId, GroupedOffenderMatches offenderMatches) {
+    private Mono<Void> postDefendantMatches(String caseId, String defendantId, GroupedOffenderMatches offenderMatches) {
```
Comment (chris-faulkner-mojdt) [inline src/main/java/uk/gov/justice/probation/courtcasematcher/restclient/CourtCaseRestClient.java:67]: Is there a reason for naming this `postDefendantMatches` from the previous `postOffenderMatches` ? It depends how you look at it, of course. They are offender matches in the sense that the entities in teh matches are offenders and not defendants but defendant matches in the sense that they are being matched to the defendant but given the fact that the parameter is named "`OffenderMatches`"  - it's now inconsistent. I'd favour keeping the existing

File: src/test/java/uk/gov/justice/probation/courtcasematcher/restclient/CourtCaseRestClientIntTest.java [hunk 3]
```diff
@@ -85,7 +87,6 @@ void setUp() {
         MockitoAnnotations.openMocks(this);
         Logger logger = (Logger) getLogger(LoggerFactory.getLogger(CourtCaseRestClient.class).getName());
         logger.addAppender(mockAppender);
-        when(featureFlags.getFlags()).thenReturn(Map.of("use-legacy-court-case-rest-client", false));
```
Comment (chris-faulkner-mojdt) [inline src/test/java/uk/gov/justice/probation/courtcasematcher/restclient/CourtCaseRestClientIntTest.java:88]: `featureFlags` is now unused in this class as is the `mono`. Remove both.

**Review by chris-faulkner-mojdt — DISMISSED** [review]
Couple of small comments

File: src/main/java/uk/gov/justice/probation/courtcasematcher/restclient/CourtCaseRestClient.java (hunk 2 above)
Comment (jhevans) [reply src/main/java/uk/gov/justice/probation/courtcasematcher/restclient/CourtCaseRestClient.java:67]: Yeah fair point, will stick with Offender. Can't recall quite why I decided Defendant was better but I think you're right

**Review by chris-faulkner-mojdt — APPROVED** [review]
LGTM

**Pull request merged**